packages/desktop/components/drawers/network-config/drawers/AddChainDrawer.svelte

```diff
@@ -17,28 +23,53 @@
         iscpEndpoint: '',
     }
 
+    function validate(): boolean {
+        if (!chain.name) {
+            nameError = localize(`${localeKey}.errors.cannotBeEmpty`)
+        }
+
+        const isHex = chain.aliasAddress.startsWith('0x')
+        if (!isHex || !/^(0x08)?[0-9a-f]{64}?(?:0[1-9]|[1-5][0-9]|6[0-4])?0{8}$/i.test(chain.aliasAddress)) {
+            aliasAddressError = localize(`${localeKey}.errors.aliasAddressMustBeHex`)
+        }
+
+        if (!isValidUrl(chain.iscpEndpoint)) {
+            iscpEndpointError = localize(`${localeKey}.errors.invalidUrl`)
+        }
+
+        if (chain.explorerUrl && !isValidUrl(chain.explorerUrl)) {
+            explorerUrlError = localize(`${localeKey}.errors.invalidUrl`)
+        }
+
+        return !!nameError || !!aliasAddressError || !!iscpEndpointError || !!explorerUrlError
+    }
+
     function handleSubmit(): void {
+        validate()
         // TODO: Fetch chainId from ISCP node before adding it to profile
     }
 </script>
 
 <add-chain-drawer class="h-full flex flex-col justify-between">
     <form id="add-chain-form" class="flex flex-col gap-3" on:submit|preventDefault={handleSubmit}>
```

Comment:
Let's go ahead and make the `AddIscpChainForm` component and put the validation logic in there - will help with the upcoming tasks around this `AddChainDrawer` :+1: